proteus/mprans/RANS2P.h

```diff
@@ -953,165 +953,31 @@ namespace proteus
-                                               const double useVF,
-                                               const double& vf,
-                                               const double& phi,
-                                               const double& p,
-                                               const double& u,
-                                               const double& v,
-                                               const double& w,
-                                               double& mom_p_acc,
-                                               double& dmom_p_acc_p,
-                                               double& mom_u_acc,
-                                               double& dmom_u_acc_u,
-                                               double& mom_v_acc,
-                                               double& dmom_v_acc_v,
-                                               double& mom_w_acc,
-                                               double& dmom_w_acc_w)
-      {
-        double H_rho, rho;
-        
-        H_rho = (1.0-useVF)*smoothedHeaviside(eps_rho,phi) + useVF*fmin(1.0,fmax(0.0,vf));
-        
-        rho = rho_0*(1.0 - H_rho) + rho_1*H_rho;
-
-        mom_p_acc = p * rho;
-        dmom_p_acc_p = rho;
-
-        mom_u_acc = u * rho;
-        dmom_u_acc_u = rho;
-
-        mom_v_acc = v * rho;
-        dmom_v_acc_v = rho;
-        
-        mom_w_acc = w * rho;
-        dmom_w_acc_w = rho;
-      } 
-      inline
-        void evaluateTPInvDensityLaplaceCoefficients(const double eps_rho,
-                                                     const double rho_0,
-                                                     const double rho_1,
-                                                     const double useVF,
-                                                     const double& vf,
-                                                     const double& phi,
-                                                     double mom_p_diff_ten[nSpace],
-                                                     double mom_u_diff_ten[nSpace],
-                                                     double mom_v_diff_ten[nSpace],
-                                                     double mom_w_diff_ten[nSpace])
-      {
-        double H_rho, rho;
-
-        H_rho = (1.0-useVF)*smoothedHeaviside(eps_rho,phi) + useVF*fmin(1.0,fmax(0.0,vf));
-
-        rho = rho_0*(1.0 - H_rho) + rho_1*H_rho;
-    
-        mom_p_diff_ten[0] = 1.0 / rho ;
-        mom_p_diff_ten[1] = 1.0 / rho ;
-        mom_p_diff_ten[2] = 1.0 / rho ;
-
-        mom_u_diff_ten[0] = 1.0 / rho ;
-        mom_u_diff_ten[1] = 1.0 / rho ;
-        mom_u_diff_ten[2] = 1.0 / rho ;
-
-        mom_v_diff_ten[0] = 1.0 / rho ;
-        mom_v_diff_ten[1] = 1.0 / rho ;
-        mom_v_diff_ten[2] = 1.0 / rho ;
 
-        mom_w_diff_ten[0] = 1.0 / rho ;
-        mom_w_diff_ten[1] = 1.0 / rho ;
-        mom_w_diff_ten[2] = 1.0 / rho ;
+	    //u momentum Hamiltonian (advection
```

Comment:
@alistairbntl tI think this is the root cause  of what @tridelat found. The additional terms for the nonconservative form still need to be initialized for conservative form.

Reply:
Indeed, I can see why this would be a problem,  I suppose without initializing these terms, the arrays could be populated with non-zero values.  This would explain the somewhat erratic failures I was seeing.